src/main/kotlin/main.kt

```diff
@@ -1,3 +1,89 @@
+class CurrencyConversion{
+    //matriz com as taxas de cambio
+    val currencyExchengeRate = arrayOf(        //AUD        BRL     EUR     IRR       USD
+                                        arrayOf(1.0,     4.3009,  0.6403,  0.0000,  0.7630),  //AUD
+                                        arrayOf(0.2324,  1.0,     0.1488,  0.0001,  0.1774),  //BRL
+                                        arrayOf(1.5616,  6.7170,  1.0,     0.0000,  1.5615),  //EUR
+                                        arrayOf(0.0000,  0.0001,  0.0000,  1.0,     0.0000),  //IRR
+                                        arrayOf(1.31021, 5.6360,  0.8380,  0.0000,  1.0) )    //USD
+
+    //possiveis moedas de conversão
+    enum class curencyCode {AUD,BRL, EUR,IRR, USD }
+
+    //atributo que recebe a moeda atual para a conversão
+    var currentCurrency: String = ""
+        set(value){
+                field = value;
+        };
+
+    //atributo que recebe a moeda para qual ira ser feita a conversão
+    var currencyDestiny:String = ""
+        set(value){
+            field = value;
+        };
+
+    //atributo que recebe o valor a ser convertido
+    var conversionValue: Double = 0.0
+        set(value){
+            if (value > 0.0){
+                field = value;
+            }else{
+                println("ENTRADA INVALIDA, NUMERO NEGATIVO")
+            }
+        };
+
+    //verificar se os codes estão listados no sistema
+    fun codeIsValid( code: String):Boolean{
```

Comment:
Só a título de curiosidade, o operador `in` também serve como um operador booleano, o que permitiria fazer algo assim:

```kotlin
fun codeIsValid(code: String): Boolean = code in listOf("AUD", "BRL", "EUR", "IRR", "USD")
```